db.go

```diff
@@ -263,7 +267,11 @@ func Open(dir string, l log.Logger, r prometheus.Registerer, opts *Options) (db
 		return nil, errors.Wrap(err, "create leveled compactor")
 	}
 
-	wlog, err := wal.New(l, r, filepath.Join(dir, "wal"))
+	segmentSize := wal.DefaultSegmentSize
+	if opts.WALSegmentSize != 0 {
```

Comment:
`WALSegmentSize` is `int`, can we have the check as `opts.WALSegmentSize > 0` instead?

Reply:
yeah probably a good idea since a negative number is not a valid setting either.

Reply:
thanks, will wait for the Prometheus 2.6 release to be done and will merge right after that. This will probably take a week.